### DISCUSSION TITLE
Please add STRAIGHT_JOIN as a modifier in EE AR code

### DISCUSSION THREAD

**Discussion by chunkitdesign** [ExpressionEngine/ExpressionEngine#3073] [Feature Requests]
I have encountered a performance issue with a specific add-on, Solspace Calendar, for several years on MariaDB. We have not found a solution until recently I came across a [doc](https://mariadb.com/kb/en/index-hints-how-to-force-query-plans/) that suggests adding `STRAIGHT_JOIN` after `SELECT` could improve the performance due to the optimizer issue.
I tested it, and the outcome is promising. On one particular query, it takes 40 seconds to load without `STRAIGHT_JOIN`, while it only takes 115 ms with it.
I want to request EE add `STRAIGHT_JOIN` as a modifier so that it will solve this particular performance issue related to Solspace Calendar.
Note: This issue is only tested on EE5 and EE6.

**Comment by chunkitdesign** [comment]
I have attached the two screenshots (From the actual add-on) for reference:

- [No `STRAIGHT_JOIN`](https://www.dropbox.com/s/088998qwqft7yhw/no_straight_join.png?dl=0)
- [With `STRAIGHT_JOIN`](https://www.dropbox.com/s/99acdsgqctzac77/with_straight_join.png?dl=0)

**Comment by robinsowell** [comment]
This might help https://stackoverflow.com/questions/71216953/the-best-way-to-do-a-mysql-straight-join-in-the-query-builder-of-codeigniter-ol

**Comment by intoeetive** [comment]
This PR adds STRAIGHT_JOIN: https://github.com/ExpressionEngine/ExpressionEngine/pull/3515
